lib/apiservers/engine/backends/cache/image_cache.go

```diff
@@ -115,3 +119,51 @@ func (c *ImageCache) GetImages() []*metadata.ImageConfig {
 
 	return result
 }
+
+func (c *ImageCache) GetImageByDigest(digest digest.Digest) *metadata.ImageConfig {
+	c.m.RLock()
+	defer c.m.RUnlock()
+
+	var imageID string
+
+	for _, image := range c.cache {
+		if strings.Contains(image.ImageID, "sha") {
+			imageID = image.ImageID
+		} else {
+			imageID = "sha256:" + image.ImageID
```

Comment:
Is this an assumption that it's a sha256? Is that enforced else where in the code - if so comment that this must be a sha256.
Are there other types it could be? If so should we iterate over those?